src/gsCore/gsMultiBasis.h

```diff
@@ -422,6 +422,38 @@ class gsMultiBasis : public gsFunctionSet<T>
         int numKnots = 1
         );
 
+    /// @brief Returns the basis that corresponds to the component
+    typename gsBasis<T>::uPtr componentBasis(patchComponent p) const
+    { return m_bases[p.patch()]->componentBasis(p); }
+
+    /// @brief Returns the basis that corresponds to the component
+    ///
+    /// @param p         The component
+    /// @param dm        The dof mapper to be used
+    /// @param indices   The row vector where the indices are stored to
+    /// @param no_lower  If true, the transfer matrix does not include parts belonging to lower-order
+    ///                  components (i.e., edges without corners or faces without corners and edges)
+    typename gsBasis<T>::uPtr componentBasis_withIndices(
+        patchComponent p,
+        const gsDofMapper& dm,
+        gsMatrix<unsigned>& indices,
+        bool no_lower = true
+    ) const;
+    
+    /// @brief Returns the bases that correspond to the components
+    ///
+    /// @param p         The component
+    /// @param dm        The dof mapper to be used
+    /// @param indices   The row vector where the indices are stored to
+    /// @param no_lower  If true, the transfer matrix does not include parts belonging to lower-order
+    ///                  components (i.e., edges without corners or faces without corners and edges)
+    std::vector<typename gsBasis<T>::uPtr> componentBasis_withIndices(
+        std::vector<patchComponent> p,
```

Comment:
std::vector<patchComponent> p
should be const reference ?

Reply:
done